client/app/scripts/liveblog-themes/module.js

```diff
@@ -0,0 +1,161 @@
+        $scope.loading = true;
+        // load only global preference for themes.
+        api.global_preferences.query({'where': {'key': 'theme'}}).then(function(data) {
+            data._items.forEach(function(item) {
+                if (item.key === 'theme') {
+                    $scope.globalTheme = item;
+                    return;
+                }
+            });
+            // load all the themes.
+            api.themes.query().then(function(data) {
+
+                // filter theme with label (without label are `generic` from inheritance)
+                $scope.themes = data._items.filter(function(theme) {
+                    if (angular.isDefined(theme.label)) {
+                        // set the priority for default theme.
+                        if ($scope.globalTheme.value === theme.name) {
+                            theme.order = 0;
+                        } else {
+                            theme.order = 1;
+                        }
+                        // create criteria to load blogs with the theme.
+                        var criteria = {
+                                source: {
+                                    query: {filtered: {filter: {term: {'theme._id': theme._id}}}}
+                                }
+                            };
+                        api.blogs.query(criteria).then(function(data) {
+                            theme.blogs = data._items;
+                        });
+                        parseTheme(theme);
+                        return true;
+                    }
+                    return false;
+                });
+            $scope.loading = false;
+            });
+        });
+
+        $scope.openThemeBlogsModal = function(theme) {
+            if (theme.blogs.length) {
+                $scope.selectedTheme = theme;
+                $scope.themeBlogsModal = true;
+            }
+        };
+
+        $scope.closeThemeBlogsModal = function(theme) {
+            if ($scope.selectedBlog) {
+                $scope.selectedBlog = false;
+            } else {
+                $scope.themeBlogsModal = false;
+            }
+        };
+
+        $scope.makeDefault = function(theme) {
+            if ($scope.globalTheme) {
+                api.global_preferences.save($scope.globalTheme, {'key': 'theme', 'value': theme.name}).then(function(data) {
+                    notify.pop();
+                    notify.info(gettext('Default theme saved'));
+                    $scope.globalTheme = data;
+                });
+            } else {
+                api.global_preferences.save({'key': 'theme', 'value': theme.name}).then(function(data) {
+                notify.pop();
+                notify.info(gettext('Default theme saved'));
+                $scope.globalTheme = data;
+            });
+
+            }
+        };
+
+        $scope.removeTheme = function(theme) {
+            api.themes.query().then(function(data) {
```

Comment:
Like in the backend, you must never retrieve all the resources if you don't need them.
First because it will overload the server and kill it :angel:, and also because there is pagination that limits the number of responses.
Then you should use the `getById()` method instead of `query()`. Or continue to use `query`, but with a parameter to filter the query.